app/services/authcore/authcore.ts

```diff
@@ -235,6 +221,27 @@ export class AuthCoreAPI {
       hrp
     )
 
+    return bech32Address
+  }
+
+  async getCosmosAddressesAndPubKeys() {
+    const pubKeys = await this.getPubKeys()
+    const addresses = pubKeys.map(pubkey => (this.getBech32Address(this.cosmosChainId,pubkey)))
+    return { addresses, pubKeys }
+  }
+
+  async getWalletConnectGetKeyResponse(
+    chainId: string,
+    { name = "" }: { name?: string } = {}
+  ) {
+    const [pubKey] = await this.getPubKeys()
+    const bech32Address = this.getBech32Address(chainId, pubKey)
+    
+    const uint8ArrayPubKey = Uint8Array.from(Buffer.from(pubKey, 'base64'))
+    let hash = CryptoJS.SHA256(CryptoJS.lib.WordArray.create(uint8ArrayPubKey)).toString()
+    hash = CryptoJS.RIPEMD160(CryptoJS.enc.Hex.parse(hash)).toString()
+    const address = Buffer.from(new Uint8Array(Buffer.from(hash, "hex"))).toString("hex")
```

Comment:
Not sure why cast the buffer to uint8[] then to buffer again
@nnkken Do you know if we have a simpler way to do this under CryptoJS?